docs/resource-collection.rst

```diff
@@ -10,11 +10,42 @@ resources (and their versions) to the user.
 The index location is set by the env/config variable ``RESOURCE_INDEX``. The
 ``RESOURCE_INDEX`` must be either a "s3://" address or a local file path. If the
 file is located on S3 it must be gzipped. The server loads this index at start
-time and refreshes it hourly. The nextstrain.org testing & production configs
-currently set this to ``s3://nextstrain-inventories/resources.json.gz``.
+time and refreshes it hourly. Resource collections can be ignored by the server
+by setting the env variable ``RESOURCE_INDEX="false"`` or by omitting it from
+your configuration JSON.
 
-Resource collections can be ignored by the server by setting the env variable
-``RESOURCE_INDEX="false"`` or by omitting it from your configuration JSON.
+
+Resource index revisions
+========================
+
+The nextstrain.org testing & production configs currently set this to
+``s3://nextstrain-inventories/resources/v<revision_number>.json.gz``.
+
+If you make any updates that changes the structure or the contents of the resource
+index JSON, then bump the ``<revision_number>`` within the configs (``env/production/config.json``)
+so that any uploads to S3 does not disrupt the production server.
+
+These updates include changes to:
+
+* any scripts within ``resourceIndexer/*``
+* ``data/manifest_core.json``
+* ``convertManifestJsonToAvailableDatasetList`` in ``src/endpoints/charon/parseManifest.js``
+* ``datasetRedirectPatterns`` in ``src/redirects.js``
+
+If you are ever unsure, it's better to just bump the revision number!
```

Comment:
_non-blocking_

How much simpler would it be if we used the commit id as the "index version" instead of a monotonically-increasing number? (I know we floated both ideas, but I don't recall discussing which to use…)

Reply:
Hmm, if we use commit id, we wouldn't run into potential version number clashes between branches. 

I thought using commit id comes with it's own weirdness if we continue to store the `RESOURCE_INDEX` in the config.json. E.g. the commit id in the `RESOURCE_INDEX` would be pointing to the parent commit id. Also, instead of updating it manually, we would probably want to set up a commit hook to automatically update the `RESOURCE_INDEX` on every commit?

Reply:
Chatted more about this with @tsibley in our 1:1 today.

If we do use the commit id as the version, then we would probably stop hard-coding the `RESOURCE_INDEX` within the config.json and just set it within the Heroku slug. It would be less obvious of what `RESOURCE_INDEX` the deployed site is using unless you download the slug or we implement https://github.com/nextstrain/nextstrain.org/issues/853. 

---

Leaving this as a future decision as we can make do with the number versions for now.

Reply:
Ah, yeah. Let's not rewrite files on every single commit! We could instead do something similar as part of the build process (e.g. bake in an updated `RESOURCE_INDEX`, or bake in a commit id in a file that's used at runtime to construct the right path, etc.). This could even be done via [Git's `ident` munging](https://git-scm.com/docs/gitattributes#_ident) (à la RCS). But let's leave that for the future, if ever.

Reply:
We can revisit this idea in https://github.com/nextstrain/nextstrain.org/issues/867